[ys112] iP

ys112 opened the pull request:
# DevGPT

> "You are too concerned with what was and what will be. There's a saying: Yesterday is history, tomorrow is a mystery, but today is a gift. That is why it is called the present." - Master Oogway ([source](https://kungfupanda.fandom.com/wiki/Oogway))

- Uses CLI
- Simple commands
- Stores 3 different types of tasks

1. Acclaimed to be the best chatbot on the market.
2. It remembers your tasks ~~forever~~
3. Runs faster than lightning McQueen! **_KACHOW_** ⚡ 

`System.out.println("DevGPT > lightningMcQueen");`

``` java
public void speed() {
    System.out.println("I am SPEED");
}
```

What can it do:
- [x] Keep track of tasks
- [x] Mark tasks done
- [x] Search for tasks
- [x] Save the tasks for next session
- [ ] Give reminders

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

ys112 committed:
Modify Duke.java

ys112 committed:
Modify Duke.java

Giving new display style to the chatbot

ys112 committed:
Modify Duke.java

Allows user to recall tasks from list

ys112 committed:
Modify Duke.java and add Task.java

User have ability to mark and unmark a task

ys112 committed:
Add Deadline.java, Event.java, Todo.java

Added support for tracking the three types of tasks

ys112 committed:
Configure automated UI testing

ys112 committed:
fix todo getting added into todo description

ys112 committed:
Implement error handling

Bot will not stop on user errors but be prompted with the valid format
to input

ys112 committed:
Add delete command

User can now delete task previously created

ys112 committed:
Modify Duke.java to enable save

Users can save their tasks and work on the same tasks in
next session.

ys112 committed:
Merge branch 'branch-Level-7'

ys112 committed:
Add ability to understand dates

User input is now handled in proper date formats

ys112 committed:
Merge branch 'branch-Level-8'

ys112 committed:
Refactor code to be more OOP

New class such as Ui, Storage, Parser, Tasklist are added to shift
the program for better encapsulation of different logic and function

ys112 committed:
Organize the classes into duke package

ys112 committed:
Merge branch 'add-gradle-support' of https://github.com/ys112/ip

ys112 committed:
Add gradle for project automations

ys112 committed:
Add JUnit to two different classes

ys112 committed:
Configure Gradle to create fat JAR

ys112 committed:
Add Javadoc comments to non-private classes and methods

Minor tweak to overwrite text file when saving

ys112 committed:
Tweak code to comply with coding standard

Checkstyle is used for consistency. JavaDoc not added as it is done
in another branch.

ys112 committed:
Fix spacing style issue in Event.java

ys112 committed:
Add find task feature

Users can now find a task using a keyword

ys112 committed:
Merge branch 'branch-Level-9'

ys112 committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/duke/Storage.java
#	src/main/java/duke/TaskList.java

ys112 committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/Deadline.java
#	src/main/java/duke/Duke.java
#	src/main/java/duke/Parser.java
#	src/main/java/duke/Task.java

ys112 committed:
Edit minor coding standard violations

ys112 committed:
Add GUI to duke using FXML and javaFX

Refactor existing code to run chatbot on GUI

ys112 committed:
Revert "Add GUI to duke using FXML and javaFX"

This reverts commit aa71a0022d1020268482c25c5f2fd9ca9e4ad05a.

ys112 committed:
Add javaFX and FXML to enable GUI in duke chatbot

ys112 committed:
Merge branch 'branch-Level-10'

sebin0817 commented on src/main/java/duke/Todo.java:
```diff
@@ -0,0 +1,24 @@
+
+    @Override
+    public String toString() {
+        return "[T]" + super.toString();
```
perhaps you could make use of getTaskType() method to return toString(), e.g. `return "[" + getTaskTypeSymbol() + "]" + super.toString();`. Other than this, good job on ensuring coding standards throughout your code!

ZhiWei1010 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,98 @@
+            tasks = new TaskList(storage.loadTasks());
+        } catch (DukeException e) {
+            ui.showError(e.getMessage());
+            tasks = new TaskList(new ArrayList<Task>());
```
Good use of plural form to distinguish multi-valued variable tasks!

ZhiWei1010 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,117 @@
+    }
+
+    private Task parseTasksFromString(String line) throws DukeException {
+        String[] split = line.split(" \\| ");
```
Perhaps split could be renamed to splits or parsedInputs to avoid confusion to a single-valued variable?

ZhiWei1010 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,127 @@
+    /**
+     * Shows error message for invalid command.
+     */
+    public String commandNotUnderstood() {
```
Perhaps method could be renamed to showInvalidCommandError()?

ZhiWei1010 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,127 @@
+            + "Now you have " + size + " tasks in the list.";
+    }
+
+    private String doubleTab() {
```
Perhaps method could be renamed to addDoubleTabs()?

ZhiWei1010 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,125 @@
+     * @throws DukeException if the date is empty
+     * @throws DateTimeParseException if the date is not in the correct format
+     */
+    public LocalDate[] getFromTo() throws DukeException, DateTimeParseException {
```
Perhaps this method could be renamed to getEventFromTo()?

ZhiWei1010 reviewed (commented):
LGTM. Just a few nits to check! Good job on following the naming conventions and following the coding standards!

ys112 committed:
Add assertion to code for ensuring code integrity

There are some parts of the code that requires assertion
to confirm that the assumptions hold
so that bugs can be spotted earlier.

Add assertions to ensure objects are not null and conditions hold in
important areas such as ensuring index given in Tasklist is
within range.

ys112 committed:
Improve code quality of chatbot

Existing code had empty default branch or using else as final option
Some are harder to read with the nesting of if else

The code is now more readable and less error prone

ys112 committed:
Merge pull request #1 from ys112/branch-A-CodeQuality

Improve code quality of chatbot

ys112 committed:
Merge branch 'master' into branch-A-Assertions

ys112 committed:
Merge pull request #2 from ys112/branch-A-Assertions

Add assertion to code for ensuring code integrity

ys112 committed:
Create gradle.yml

ys112 committed:
Fix extra assert in code

ys112 committed:
Add assertion to Duke.java

Minor fix for saving and retrieving tasks from same text file

ys112 committed:
Fix storage not loading tasks

ys112 committed:
Add reminder to inform user about the given deadline

Some users prefer that the chatbot notifies them about their upcoming
deadline hence there will be a popup window to alert them.

Change some variable names to more verbose ones to improve readability
and made default path to reduce error prone code.

ys112 committed:
Add user guide

The user guide aims to help users navigate the chatbot and understand
the features the chatbot can provide.

Resolve a few cosmetics issues for displaying title and logo in GUI
window and generated jar file in this commit.

ys112 committed:
Update to user guide

ys112 committed:
Fix welcome message on chatbot run

Commit contains cosmetic fix for dialog positioning.

ys112 committed:
Fix chatbot not exiting

Commit contains corrections for user guide.
